engines/sci/sound/audio32.cpp

```diff
@@ -0,0 +1,920 @@
+/* ScummVM - Graphic Adventure Engine
+ *
+ * ScummVM is the legal property of its developers, whose names
+ * are too numerous to list here. Please refer to the COPYRIGHT
+ * file distributed with this source distribution.
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+ *
+ */
+
+#include "sci/sound/audio32.h"
+#include "audio/audiostream.h"      // for SeekableAudioStream
+#include "audio/decoders/raw.h"     // for makeRawStream, RawFlags::FLAG_16BITS
+#include "audio/decoders/wave.h"    // for makeWAVStream
+#include "audio/rate.h"             // for RateConverter, makeRateConverter
+#include "audio/timestamp.h"        // for Timestamp
+#include "common/config-manager.h"  // for ConfMan
+#include "common/endian.h"          // for MKTAG
+#include "common/memstream.h"       // for MemoryReadStream
+#include "common/str.h"             // for String
+#include "common/stream.h"          // for SeekableReadStream
+#include "common/system.h"          // for OSystem, g_system
+#include "common/textconsole.h"     // for warning
+#include "common/types.h"           // for Flag::NO
+#include "engine.h"                 // for Engine, g_engine
+#include "sci/engine/vm_types.h"    // for reg_t, make_reg, NULL_REG
+#include "sci/resource.h"           // for ResourceId, ResourceType::kResour...
+#include "sci/sci.h"                // for SciEngine, g_sci, getSciVersion
+#include "sci/sound/decoders/sol.h" // for makeSOLStream
+
+namespace Sci {
+
+bool detectSolAudio(Common::SeekableReadStream &stream) {
+	const size_t initialPosition = stream.pos();
+
+// TODO: Resource manager for audio resources reads past the
+// header so even though this is the detection algorithm
+// in SSCI, ScummVM can't use it
```

Comment:
The SCI resource manager checks the type of each resource that is loaded. Thus, the resource type check does happen in ScummVM, but it occurs inside the resource manager, not in the SOL audio detection function